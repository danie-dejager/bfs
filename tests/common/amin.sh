cd "$TEST"

now=$(epoch_time)

"$XTOUCH" -at "@$((now - 60 * 60))" one_hour_ago
"$XTOUCH" -at "@$((now - 121))" two_minutes_ago
"$XTOUCH" -at "@$((now - 61))" one_minute_ago
"$XTOUCH" -at "@$((now - 30))" thirty_seconds_ago
"$XTOUCH" -at "@$((now + 60))" in_one_minute
"$XTOUCH" -at "@$((now + 60 * 60))" in_one_hour

bfs_diff . -mindepth 1 \
    \( -amin -1 -exec printf -- '-amin -1: %s\n' {} \; -o -true \) \
    \( -amin  1 -exec printf -- '-amin  1: %s\n' {} \; -o -true \) \
    \( -amin +1 -exec printf -- '-amin +1: %s\n' {} \; -o -true \)
