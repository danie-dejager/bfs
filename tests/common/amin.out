-amin  1: ./one_minute_ago
-amin +1: ./one_hour_ago
-amin +1: ./two_minutes_ago
-amin -1: ./in_one_hour
-amin -1: ./in_one_minute
-amin -1: ./thirty_seconds_ago
